src/RewardsDistributor.sol

```diff
@@ -96,10 +102,17 @@ contract RewardsDistributor is IRewardDistributor {
                 "Collateral does not match the rewards token"
             );
         }
+
+        // amount_ is in payout token decimals precision, adjust actual distribution amount to 18 decimals that core is making its calculations in
+        // this is necessary to avoid rounding issues when doing actual payouts
+        uint256 systemPrecision = 10 ** 18;
+        uint256 distributorPrecision = 10 ** IERC20(payoutToken).decimals();
+        uint256 adjustedAmount = (amount_ / distributorPrecision) * systemPrecision;
```

Comment:
what if the distributed amount is 0.1 tokens?

Reply:
there is no such thing as 0.1 token. For 18 dec tokens it will be `1e17`, still round numbers.

Reply:
Added tests for 0.001 token distribution and payout